src/Datadog.Trace/Tracer.cs

```diff
@@ -45,15 +46,23 @@ public Tracer()
         /// or null to use the default configuration sources.
         /// </param>
         public Tracer(TracerSettings settings)
-            : this(settings, agentWriter: null, sampler: null, scopeManager: null)
+            : this(settings, agentWriter: null, sampler: null, scopeManager: null, dogStatsdClient: null)
         {
         }
 
-        internal Tracer(TracerSettings settings, IAgentWriter agentWriter, ISampler sampler, IScopeManager scopeManager)
+        internal Tracer(TracerSettings settings, IAgentWriter agentWriter, ISampler sampler, IScopeManager scopeManager, IDogStatsd dogStatsdClient)
         {
             // fall back to default implementations of each dependency if not provided
             Settings = settings ?? TracerSettings.FromDefaultSources();
-            _agentWriter = agentWriter ?? new AgentWriter(new Api(Settings.AgentUri));
+
+            // only set DogStatsdClient if tracer metrics are enabled
+            if (Settings.TracerMetricsEnabled)
+            {
+                DogStatsdClient = dogStatsdClient ?? CreateDogStatsdClient(Settings);
```

Comment:
I so love the C# language for giving us the ability to avoid nulls so easily.

Reply:
So the idea behind using the null conditional is that metrics are enabled if the dog stats client is not null?

Reply:
The null conditional itself doesn't enable or disable metrics. It's used to create a default `IDogStatsdClient` implementation _if metrics are enabled_ but a `IDogStatsdClient` is not provided as a constructor argument. Creating a "default" implementation is what we already do if any of the other constructor arguments are null.